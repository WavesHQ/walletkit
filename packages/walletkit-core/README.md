[![npm](https://img.shields.io/npm/v/@waveshq/walletkit-core)](https://www.npmjs.com/package/@waveshq/walletkit-core/v/latest)

# @waveshq/walletkit-core

Contains all core logic for wallet apps
