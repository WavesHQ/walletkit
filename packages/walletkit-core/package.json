{
  "private": false,
  "name": "@waveshq/walletkit-core",
  "version": "0.0.0",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "tsc -b ./tsconfig.build.json",
    "test": "jest --maxWorkers=100%",
    "lint": "eslint .",
    "clean": "rm -rf dist"
  },
  "jest": {
    "preset": "@birthdayresearch/sticky-jest"
  },
  "dependencies": {
    "bignumber.js": "^9.1.1",
    "ethers": "^5.7.2"
  },
  "devDependencies": {
    "@types/react": "18.2.0",
    "@birthdayresearch/sticky-jest": "^0.6.2"
  }
}
