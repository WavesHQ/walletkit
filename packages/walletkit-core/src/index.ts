export * from "./api";
export * from "./testcontainers";
export * from "./website";
