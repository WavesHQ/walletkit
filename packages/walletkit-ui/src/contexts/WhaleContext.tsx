import { WhaleApiClient, WhaleRpcClient } from "@defichain/whale-api-client";
import {
  newOceanOptions,
  newWhaleAPIClient,
  newWhaleRpcClient,
} from "@waveshq/walletkit-core";
import React, { createContext, useContext, useMemo } from "react";

import { useNetworkContext } from "./NetworkContext";

const WhaleApiClientContext = createContext<{
  whaleAPI: WhaleApiClient;
  whaleRPC: WhaleRpcClient;
}>(undefined as any);

export function useWhaleApiClient(): WhaleApiClient {
  return useContext(WhaleApiClientContext).whaleAPI;
}

export function useWhaleRpcClient(): WhaleRpcClient {
  return useContext(WhaleApiClientContext).whaleRPC;
}

export function WhaleProvider({
  children,
}: React.PropsWithChildren<any>): JSX.Element | null {
  const { network } = useNetworkContext();
  const client = useMemo(
    () => ({
      whaleAPI: newWhaleAPIClient(newOceanOptions(network)),
      whaleRPC: newWhaleRpcClient(newOceanOptions(network)),
    }),
    [network]
  );

  return (
    <WhaleApiClientContext.Provider value={client}>
      {children}
    </WhaleApiClientContext.Provider>
  );
}
