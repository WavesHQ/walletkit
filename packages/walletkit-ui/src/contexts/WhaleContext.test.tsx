/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import { EnvironmentNetwork } from "@waveshq/walletkit-core";
import React from "react";

import * as NetworkContext from "./NetworkContext";
import { useWhaleApiClient, WhaleProvider } from "./WhaleContext";

const networkContext = jest.spyOn(NetworkContext, "useNetworkContext");

describe("Whale Context test", () => {
  const networkDetails = [
    {
      url: "https://ocean.defichain.com",
      network: "mainnet",
      name: EnvironmentNetwork.MainNet,
    },
    {
      url: "https://testnet.ocean.jellyfishsdk.com",
      network: "testnet",
      name: EnvironmentNetwork.TestNet,
    },
    {
      url: "https://playground.jellyfishsdk.com",
      network: "regtest",
      name: EnvironmentNetwork.RemotePlayground,
    },
    {
      url: "http://localhost:19553",
      network: "regtest",
      name: EnvironmentNetwork.LocalPlayground,
    },
    {
      url: "http://devnet.ocean.jellyfishsdk.com:3000",
      network: "devnet",
      name: EnvironmentNetwork.DevNet,
    },
  ];

  networkDetails.forEach((networkDetail) => {
    describe(`Whale Context test for ${networkDetail.name}`, () => {
      it("should match snapshot", () => {
        networkContext.mockImplementation(
          () =>
            ({
              network: networkDetail.name,
            } as any)
        );
        function WhaleProviderComponent(): JSX.Element {
          const client = useWhaleApiClient() as any;
          expect(Object.keys(client)).toEqual(
            expect.arrayContaining([
              "options",
              "rpc",
              "address",
              "poolpairs",
              "transactions",
              "tokens",
              "masternodes",
              "blocks",
              "oracles",
              "prices",
              "stats",
              "rawtx",
              "fee",
              "loan",
            ])
          );
          const options = {
            version: "v0",
            timeout: 60000,
            url: networkDetail.url,
            network: networkDetail.network,
          };
          expect(client.options).toMatchObject(options);
          return (
            <div>
              <span>{JSON.stringify(client.options)}</span>
            </div>
          );
        }

        const rendered = render(
          <WhaleProvider>
            <WhaleProviderComponent />
          </WhaleProvider>
        );
        expect(rendered).toMatchSnapshot();
      });
    });
  });
});
