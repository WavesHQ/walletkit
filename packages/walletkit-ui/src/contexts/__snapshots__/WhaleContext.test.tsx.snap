// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Whale Context test Whale Context test for DevNet should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <span>
          {"url":"http://devnet.ocean.jellyfishsdk.com:3000","timeout":60000,"version":"v0","network":"devnet"}
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <span>
        {"url":"http://devnet.ocean.jellyfishsdk.com:3000","timeout":60000,"version":"v0","network":"devnet"}
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Whale Context test Whale Context test for Local should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <span>
          {"url":"http://localhost:19553","timeout":60000,"version":"v0","network":"regtest"}
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <span>
        {"url":"http://localhost:19553","timeout":60000,"version":"v0","network":"regtest"}
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Whale Context test Whale Context test for MainNet should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <span>
          {"url":"https://ocean.defichain.com","timeout":60000,"version":"v0","network":"mainnet"}
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <span>
        {"url":"https://ocean.defichain.com","timeout":60000,"version":"v0","network":"mainnet"}
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Whale Context test Whale Context test for Playground should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <span>
          {"url":"https://playground.jellyfishsdk.com","timeout":60000,"version":"v0","network":"regtest"}
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <span>
        {"url":"https://playground.jellyfishsdk.com","timeout":60000,"version":"v0","network":"regtest"}
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Whale Context test Whale Context test for TestNet should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <span>
          {"url":"https://testnet.ocean.jellyfishsdk.com","timeout":60000,"version":"v0","network":"testnet"}
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <span>
        {"url":"https://testnet.ocean.jellyfishsdk.com","timeout":60000,"version":"v0","network":"testnet"}
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
